### PR TITLE
log redaction and packege image for deployment

### PR DESCRIPTION
**Pull request by cob16**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,7 +7,7 @@ SHELL = /bin/bash
 
 PYTHON_VERSION = $(shell head -1 .python-version)
 
-POETRY = docker run \
+DOCKER = docker run \
```
Comment (wunzeco) [inline Makefile:10]: There is a `poetry run` command at the end. I think `POETRY` correctly reflects the intent.

File: bitwarden_manager/bitwarden_manager.py [hunk 2]
```diff
@@ -1,32 +1,17 @@
...
-        self._logger.info(self.get_ldap_username())
+        self._secretsmanager = AwsSecretsManagerClient(secretsmanager_client=boto3.client("secretsmanager"))
 
     def get_ldap_username(self) -> str:
         return self._secretsmanager.get_secret_value("/bitwarden/ldap-username")
 
     def get_ldap_password(self) -> str:
         return self._secretsmanager.get_secret_value("/bitwarden/ldap-password")
 
-    def _configure_logging(self, log_level: str) -> logging.Logger:
```
Comment (baarkerlounger) [inline bitwarden_manager/bitwarden_manager.py:22]: Why is the logging all getting removed?

File: tests/test_bitwarden_manager.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+from unittest import mock
+from unittest.mock import Mock, call
+
+from bitwarden_manager.bitwarden_manager import BitwardenManager
+
+
+@mock.patch("boto3.client")
+def test_get_credentails(mock_secretsmanager):
```
Comment (baarkerlounger) [inline tests/test_bitwarden_manager.py:8]: Typo on `credentials`

File: Dockerfile [hunk 4]
```diff
@@ -7,20 +7,22 @@ RUN pip install \
...
-RUN poetry install --with=dev
+RUN poetry install --no-root --with=dev
+COPY . .
 
 
 #---- lambda ----
 FROM base AS lambda
 
+COPY bitwarden_manager .
+COPY main.py .
```
Comment (wunzeco) [inline Dockerfile:25]: where is `main.py`?

File: Makefile (hunk 1 above)
Comment (cob16) [reply Makefile:10]: that's a good point, it did confuse me at first when looking at other makefiles. https://github.com/hmrc/platsec-aws-scanner/blob/main/Makefile would `POETRY_DOCKER` be a more appropriate name?

File: bitwarden_manager/bitwarden_manager.py (hunk 2 above)
Comment (cob16) [reply bitwarden_manager/bitwarden_manager.py:22]: Next pr adds our own logging class that does secrets redaction

File: Dockerfile (hunk 4 above)
Comment (baarkerlounger) [reply Dockerfile:25]: Did you mean `app.py`?

File: mypy.ini [hunk 5]
```diff
@@ -1,4 +1,5 @@
 [mypy]
+exclude = tests
```
[the remark this answers is not in the document]
Comment (cob16) [reply mypy.ini:2]: Personally, adding type checks to unit tests can take a lot of time for not much benefit compared to prod code, happy to remove if you like

File: Makefile [hunk 6]
```diff
@@ -29,29 +29,29 @@ init:
...
+	@$(DOCKER) bandit -c bandit.yaml -r -q .
 
 python-test: python
-	@$(POETRY) pytest \
+	@$(DOCKER) pytest \
 		-v \
 		-p no:cacheprovider \
 		--no-header \
-		--cov=. \
+		--cov=bitwarden_manager \
```
Comment (wunzeco) [inline Makefile:51]: 👍🏾

File: bitwarden_manager/bitwarden_manager.py (hunk 2 above)
Comment (wunzeco) [reply bitwarden_manager/bitwarden_manager.py:22]: Same question as @baarkerlounger 

Also, I need to demonstrate reading of a secrets from secretsmanager store. Refactoring `run()` and `__configure_logging()` away does not work for me.

File: mypy.ini (hunk 5 above)
[the remark this answers is not in the document]
Comment (wunzeco) [reply mypy.ini:2]: +1

File: tests/test_app.py [hunk 7]
```diff
@@ -0,0 +1,8 @@
+from unittest import mock
+
+from app import handler
+
+
+@mock.patch("boto3.client")
+def test_handler(boto3_mock):
```
Comment (wunzeco) [inline tests/test_app.py:7]: missing type?

File: Dockerfile (hunk 4 above)
Comment (cob16) [reply Dockerfile]: ah good spot will fix

File: bitwarden_manager/bitwarden_manager.py (hunk 2 above)
Comment (cob16) [reply bitwarden_manager/bitwarden_manager.py:22]: @wunzeco I have added separate tests for this. but please note the next pr is going to set up logging and run for actual use

File: tests/test_bitwarden_manager.py (hunk 3 above)
Comment (cob16) [reply tests/test_bitwarden_manager.py]: fixed

File: bitwarden_manager/bitwarden_manager.py (hunk 2 above)
Comment (cob16) [reply bitwarden_manager/bitwarden_manager.py:22]: @wunzeco this is the test i mentioned https://github.com/hmrc/bitwarden-manager/pull/10/files#diff-2e9c182581ec32ffc508c6490309eed5922048001a9f1989c3d8c9bb9cc40253R8

**Review by baarkerlounger — APPROVED** [review]

File: bitwarden_manager/bitwarden_manager.py (hunk 2 above)
Comment (wunzeco) [reply bitwarden_manager/bitwarden_manager.py:22]: 👍🏾

**Review by wunzeco — APPROVED** [review]

File: bitwarden_manager/redacting_formatter.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import logging
+import re
+from typing import List
+
+PATTERNS = [r"organization.[\w-]{36}", r"\b\w{30}\b"]
```
Comment (baarkerlounger) [inline bitwarden_manager/redacting_formatter.py:5]: Think it's worth a comment explaining what pattern the regex is intended to match

**Pull request merged**